src/transformers/models/clip/modeling_clip.py

```diff
@@ -1304,3 +1305,56 @@ def forward(
             hidden_states=vision_outputs.hidden_states,
             attentions=vision_outputs.attentions,
         )
+
+
+@add_start_docstrings(
+    """
+    CLIP Model with a a linear layer on top of the CLIPOutput.image_embeds.
+    """,
+    CLIP_START_DOCSTRING,
+)
+class CLIPForImageClassification(CLIPPreTrainedModel):
+    """
+    Repo with custom implementation: https://github.com/Andron00e/CLIPForImageClassification
+    Pre-trained model on hub: https://huggingface.co/Andron00e/CLIPForImageClassification-v1
+    """
+    def __init__(self, config: CLIPConfig, num_labels: int):
+        super().__init__(config)
+        self.clip = CLIPModel.from_pretrained("openai/clip-vit-base-patch32")
+        for p in self.clip.parameters():
+            p.requires_grad=False
+        self.num_labels = num_labels
+        self.head = nn.Linear(self.config.projection_dim, num_labels)
+        self.processor = CLIPProcessor.from_pretrained("openai/clip-vit-base-patch32")
+
+    def forward(
+        self,
+        input_ids: Optional[torch.LongTensor] = None,
+        pixel_values: Optional[torch.FloatTensor] = None,
+        attention_mask: Optional[torch.Tensor] = None,
+        position_ids: Optional[torch.LongTensor] = None,
+        return_loss: Optional[bool] = None,
+        output_attentions: Optional[bool] = None,
+        output_hidden_states: Optional[bool] = None,
+        return_dict: Optional[bool] = None,
+        labels: Optional[torch.Tensor] = None,
+        ):
+        clip_outputs = self.clip(
+            input_ids=input_ids,
+            pixel_values=pixel_values,
+            attention_mask=attention_mask,
+            position_ids=position_ids,
+            return_loss=return_loss,
+            output_attentions=output_attentions,
+            output_hidden_states=output_hidden_states,
+            return_dict=return_dict,
+        )
+
+        logits = self.head(clip_outputs.image_embeds)
+
+        loss = None
```

Comment:
here, the same logic as here needs to be defined: https://github.com/huggingface/transformers/blob/4d4febb7aa3f49c29d7f4fc29bbc1760edd583b6/src/transformers/models/dinov2/modeling_dinov2.py#L721-L757